src/rocm.jl

```diff
@@ -1,11 +1,11 @@
 import .AMDGPU
 
 function Base.cconvert(::Type{MPIPtr}, A::AMDGPU.ROCArray{T}) where T
-    Base.cconvert(Ptr{T}, A.buf.ptr) # returns DeviceBuffer
+    A
```

Comment:
What was wrong here @simonbyrne ?

Reply:
`A.buf.ptr` is a raw pointer, so the object could be cleaned up by GC. It also loses the `offset` argument required for views.

Reply:
Thanks! TIL